app/scripts/migrations/165.ts

```diff
@@ -0,0 +1,97 @@
+import { cloneDeep } from 'lodash';
+import { hasProperty, isObject } from '@metamask/utils';
+import {
+  CHAIN_IDS,
+  getFailoverUrlsForInfuraNetwork,
+} from '../../../shared/constants/network';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+export const version = 165;
+
+/**
+ * This migration adds Base network to the `NetworkController` as it is enabled by default.
+ * - It modifies the `NetworkController.state.networkConfigurationsByChainId` to include Base network.
+ *
+ * @param originalVersionedData - The versioned extension state.
+ * @returns The updated versioned extension state with base network added
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+
+  versionedData.data = transformState(versionedData.data);
+
+  return versionedData;
+}
+
+function transformState(
+  state: Record<string, unknown>,
+): Record<string, unknown> {
+  if (!hasProperty(state, 'NetworkController')) {
+    console.warn(`Migration ${version}: NetworkController not found.`);
+    return state;
```

Comment:
We are not emitting a sentry error as this might be a legitimate missing controller from old upgrades.